lib/puppet/type/netapp_security_login.rb

```diff
@@ -0,0 +1,32 @@
+Puppet::Type.newtype(:netapp_security_login) do
+  @doc = "A user account associated with the specified application and authentication method. A new user account can be created with user name as the Active Directory group name. This user account gives access to users belonging to the specified Active Directory group."
+
+  apply_to_device
+
+  ensurable
+
+  newparam(:name) do
+    desc "A composite key made up from application:authentication_method:username:vserver eg ssh:password:vsadmin:vserver01"
+    isnamevar
+  end
+
+  newproperty(:password) do
+    desc "Password for the user account. This is ignored for creating snmp users. This is required for creating non-snmp users."
```

Comment:
Can the password be read from the system? Perhaps it needs to be a parameter (ie, if we can't read if from the system then we don't know when to update it, so it has to be a parameter instead of a property).

Users should be able to force a password update by changing the comment (which will cause a resource sync) for a hacky workaround.